src/bidiTab/mapperTabPage.ts

```diff
@@ -22,68 +22,70 @@ declare global {
-    }
-    const typeLogContainer = this.#findOrCreateTypeLogContainer(logType);
+/**
+ * The following piece of HTML should be added to the `debug` element:
+ *
+ * <div class="divider"></div>
+ * <div class="item">
+ * <h3>${name}</h3>
+ * <div id="${name}_log" class="log">
+ */
+function findOrCreateTypeLogContainer(logType: string): HTMLElement {
+  const containerId = logType + '_log';
 
-    // This piece of HTML should be added:
-    /*
-      <div class="pre">...log message...</div>
-    */
-    const lineElement = document.createElement('div');
-    lineElement.className = 'pre';
-    lineElement.textContent = messages.join(', ');
-    typeLogContainer.appendChild(lineElement);
+  const existingContainer = document.getElementById(containerId);
+  if (existingContainer) {
+    return existingContainer;
   }
 
-  // This piece of HTML should be added to the `debug` element:
-  /*
-      <div class="divider"></div>
-      <div class="item">
-        <h3>${name}</h3>
-        <div id="${name}_log" class="log">
-  */
-  static #findOrCreateTypeLogContainer(logType: string) {
-    const containerId = logType + '_log';
+  const debugElement = document.getElementById('details')!;
 
-    const existingContainer = document.getElementById(containerId);
-    if (existingContainer) {
-      return existingContainer;
-    }
+  const divider = document.createElement('div');
+  divider.className = 'divider';
+  debugElement.appendChild(divider);
 
-    const debugElement = document.getElementById('details')!;
+  const htmlItem = document.createElement('div');
+  htmlItem.className = 'item';
+  htmlItem.innerHTML = `<h3>${logType}</h3><div id="${containerId}" class="log"></div>`;
+  debugElement.appendChild(htmlItem);
 
-    const divider = document.createElement('div');
-    divider.className = 'divider';
-    debugElement.appendChild(divider);
+  return document.getElementById(containerId)!;
+}
 
-    const htmlItem = document.createElement('div');
-    htmlItem.className = 'item';
-    htmlItem.innerHTML = `<h3>${logType}</h3><div id="${containerId}" class="log"></div>`;
-    debugElement.appendChild(htmlItem);
+export function log(logType: LogType, ...messages: unknown[]) {
+  // If run not in browser (e.g. unit test), do nothing.
+  if (!globalThis.document?.documentElement) {
+    return;
+  }
+  const typeLogContainer = findOrCreateTypeLogContainer(logType);
+
+  /*
+   * The following piece of HTML should be added:
+   *
+   * <div class="pre">...log message...</div>
+   */
+  const lineElement = document.createElement('div');
+  lineElement.className = 'pre';
+  lineElement.textContent = messages.join(', ');
+  typeLogContainer.appendChild(lineElement);
+}
 
-    return document.getElementById(containerId)!;
+export class MapperTabPage {
+  static generatePage() {
+    // If not in a browser (e.g. unit test), do nothing.
+    if (!globalThis.document?.documentElement) {
+      return;
+    }
+    window.MapperTabPage = MapperTabPage;
```

Comment:
I could get rid of `generatePage` too but I am not sure what to do with this assignment. Thoughts?